20230605-foundations-ggplot2/slides.qmd

```diff
@@ -0,0 +1,867 @@
+```
+
+![](img/penguins.png){width=350 fig-align="center" fig-alt="Cartoon image of three Chinstrap, Gentoo, and Adelie penguins."}
+:::
+
+::: {.column width="60%"}
+A plot contains:
+
+::: incremental
+
+- Data and aesthetic mapping
+
+- Geometric object(s)
+
+- Statistical transformation(s)
+
+- Scales 
+
+- Coordinate system
+
+- Optional faceting or theming
+
+:::
+
+:::
+:::
+
+## A layered grammar of graphics
+
+::: columns
+::: {.column width="40%"}
+
+```{r}
+#| warning: false
+#| echo: false
+#| fig-width: 4
+#| fig-height: 4
+
+library(palmerpenguins)
+library(ggplot2)
+
+ggplot(penguins, aes(x = bill_length_mm, y = bill_depth_mm, 
+                     size = body_mass_g,
+                     color = species)) +
+  geom_point(alpha = 0.5) +
+  theme_classic(base_size = 16) +
+  scale_color_manual(values = c("#FF8100", "#C75BCB", "#057575")) +
+  guides(size = "none",
+         color = "none")
+```
+
+![](img/penguins.png){width=350 fig-align="center" fig-alt="Cartoon image of three Chinstrap, Gentoo, and Adelie penguins."}
+:::
+
+::: {.column width="60%"}
+A layer contains:
+
+::: incremental
+
+- Data with aesthetic mapping
+
+- A statistical transformation, or *stat*
+
+- A geometric object, or *geom*
+
+- A position adjustment 
+
+:::
+
+:::
+:::
+
+## Mapping data to aesthetics 
+
+Data can be added to either the entire ggplot object or a particular layer. 
+
+Input data must be a dataframe in 'tidy' format:
+
+::: incremental
+
+ - every column is a variable
+ 
+ - every row is an observation
+ 
+ - every cell is a single value
+ 
+:::
+
+::: callout-note
+
+See ["Tidy Data"](https://vita.had.co.nz/papers/tidy-data.html) by Wickham (2014) and the associated 
+[vignette](https://cran.r-project.org/web/packages/tidyr/vignettes/tidy-data.html)
+:::
+
+## Mapping data to aesthetics 
+
+Subset of original data:
+
+```{r}
+#| echo: false
+example <- penguins[c(1:2, 160:161, 283:284), c(1, 3:4, 6)]
+example
+```
+
+## Mapping data to aesthetics 
+
+::: columns
+::: {.column width="40%"}
+
+```{r}
+#| warning: false
+#| echo: false
+#| fig-width: 4
+#| fig-height: 4
+library(palmerpenguins)
+library(ggplot2)
+
+ggplot(penguins, aes(x = bill_length_mm, y = bill_depth_mm, 
+                     size = body_mass_g,
+                     color = species)) +
+  geom_point(alpha = 0.5) +
+  theme_classic(base_size = 16) +
+  scale_color_manual(values = c("#FF8100", "#C75BCB", "#057575")) +
+  guides(size = "none",
+         color = "none")
+```
+
+```{r}
+#| warning: false
+#| eval: false
+aes(x = bill_length_mm,
+    y = bill_depth_mm,
+    size = body_mass_g,
+    color = species)
+```
+:::
+
+::: {.column width="60%"}
+Variables mapped to aesthetic:
+
+```{r}
+#| echo: false
+library(dplyr)
+example <- penguins[c(1:2, 160:161, 283:284), c(1, 3:4, 6)]
+example %>%
+  rename(x = bill_length_mm,
+         y = bill_depth_mm,
+         Size = body_mass_g,
+         Color = species)
+```
+
+:::
+:::
+
+## Where to specify aesthetics?
```

Comment:
This slide is a little crowded.  Might not need the plot and can focus on code chunks instead